colcon_core/shell/__init__.py

```diff
@@ -266,24 +267,29 @@ def get_shell_extensions():
     :rtype: dict
     """
     output = await check_output(cmd, cwd=cwd, shell=shell)
-    env = {}
+    env = OrderedDict()
     for line in output.splitlines():
         line = line.rstrip()
         if not line:
             continue
         encoding = locale.getpreferredencoding()
         try:
-            parts = line.decode(encoding).split('=', 1)
+            line = line.decode(encoding)
         except UnicodeDecodeError:
             line_replaced = line.decode(encoding=encoding, errors='replace')
             logger.warning(
                 'Failed to decode line from the environment using the '
                 "encoding '{encoding}': {line_replaced}".format_map(locals()))
             continue
-        if len(parts) != 2:
-            # skip lines which don't contain an equal sign
-            continue
-        env[parts[0]] = parts[1]
+        parts = line.split('=', 1)
+        if len(parts) == 2 and re.match('^[a-zA-Z_][a-zA-Z0-9_%]*$', parts[0]):
```

Comment:
I just verified that in Windows, environment vars can start with `[0-9_%]`. I think that is why the test is failing.

Reply:
Yeah, I added a message to the assert that you just removed, and that line is a valid entry. We shouldn't ignore it: https://ci.appveyor.com/project/ScottKLogan/colcon-core

Reply:
```suggestion
        if len(parts) == 2 and re.match('^[a-zA-Z0-9_%]*$', parts[0]):
```

Reply:
The key needs to be at least one character. So the `*` needs to become a `+`.

Reply:
Good catch.